pkg/provisioners/kafka/controller/channel/provider.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package channel
+
+import (
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+)
+
+const (
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "kafka-provisioner-channel-controller"
+)
+
+type reconciler struct {
+	client     client.Client
+	restConfig *rest.Config
+	recorder   record.EventRecorder
+	log        logr.Logger
```

Comment:
I think the preference is for all new code to use zap loggers.

Reply:
It's just the interface, the logger implementation is stil ZapLogger. Isn't that fine?

Reply:
My mistake. I think you are right, it probably is fine.

Reply:
what is the benefit of logr?

Reply:
also, if you store the logger inside the reconciler vs passing a context and reconstructing the zap logger you do not get to benefit from the logger decoration we are intending to use in the future

Reply:
Done